src/app/services/selector_matcher.js

```diff
@@ -125,14 +127,33 @@ function getNodesByPath(elements, path) {
     return nodes;
 }
 
+function getNodesByFile(elements, path) {
+    var nodes = [];
+
+    _.each(elements, function(node) {
+        var path_parts = node.data.original_file_path.split("/");
+        var fname = _.last(path_parts);
+        if (fname == path) {
+            nodes.push(node.data);
+        }
+    })
+    return nodes;
+}
+
 function getNodesByImplicitSelection(elements, selector) {
     var fqn_matched = getNodesByFQN(elements, selector);
     var path_matched = getNodesByPath(elements, selector);
 
+    var file_matched = [];
+    if (selector.toLowerCase().endsWith('.sql')) {
+        file_matched = getNodesByFile(elements, selector);
+    }
```

Comment:
We can replace `var` with `const` if we conditionally assign here
```suggestion
    const fileMatched = selector.toLowerCase().endsWith('.sql')
        ? getNodesByFile(elements, selector)
        : [];
```